### PR TITLE
python3Packages.onnx, python3Packages.skl2onnx: fix build

### PR DESCRIPTION
**Pull request by applePrincess**
###### Description of changes
Fix build failure by relaxing dependency version requirements.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by applePrincess** [comment]
CC: @acairncross for testing python3Packages.onnx

**Comment by applePrincess** [comment]
Result of `nixpkgs-review pr 197194` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>python310Packages.ffcv</li>
    <li>python310Packages.onnx</li>
    <li>python310Packages.onnxconverter-common</li>
    <li>python310Packages.pytorch-pfn-extras</li>
    <li>python310Packages.skl2onnx</li>
    <li>python39Packages.ffcv</li>
    <li>python39Packages.onnx</li>
    <li>python39Packages.onnxconverter-common</li>
    <li>python39Packages.pytorch-pfn-extras</li>
    <li>python39Packages.skl2onnx</li>
  </ul>
</details>

**Comment by FRidh** [comment]
Could you check whether it is possible to enable tests for skl2onnx.

**Comment by applePrincess** [comment]
I still get Segmentation fault.

File: pkgs/development/python-modules/onnx/default.nix [hunk 1]
```diff
@@ -48,6 +48,8 @@ buildPythonPackage rec {
     substituteInPlace tools/protoc-gen-mypy.sh.in \
       --replace "/bin/bash" "${bash}/bin/bash"
 
+    substituteInPlace  requirements.txt \
+      --replace "protobuf >= 3.12.2, <= 3.20.1" ""
```
Comment (acairncross) [inline pkgs/development/python-modules/onnx/default.nix:52]: I think
```suggestion
      --replace "protobuf >= 3.12.2, < 4.0.0" ""
```
would be safer. In fact I see that that's what they've done upstream: https://github.com/onnx/onnx/pull/4535. (Ideally we would use the upstream patch using `fetchpatch` but I get some issues with the PyPI line endings when applying the patch.)

Comment (applePrincess) [reply pkgs/development/python-modules/onnx/default.nix]: fixed!

**Comment by FRidh** [comment]
Note there is also the pythonRelaxDepsHook for relaxing these constraints.

File: pkgs/development/python-modules/onnx/default.nix (hunk 1 above)
Comment (applePrincess) [reply pkgs/development/python-modules/onnx/default.nix]: I thought I fixed, but I could not build onnx (due to conflicting constraints). So I just use `pythonRelaxDepsHook` for both onnx and skl2onnx (as suggested https://github.com/NixOS/nixpkgs/pull/197194#issuecomment-1288060622)

**Review by acairncross — APPROVED** [review]

**Pull request merged**